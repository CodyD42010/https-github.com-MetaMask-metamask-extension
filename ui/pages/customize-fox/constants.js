export const POLISH_OPTIONS = [
  {
    value: 'generative',
    name: 'Generative',
  },
  {
    value: 'ai',
    name: 'AI',
  },
  {
    value: 'editorSelection',
    name: "Editor's selection",
  },
  {
    value: 'default',
    name: 'Your favourite fox',
  },
];

export const EditorSelectionOptions = [
  {
    value: '1',
    name: 'Rose Hue',
  },
  {
    value: '2',
    name: 'Cocoa Chic',
  },
  {
    value: '3',
    name: 'Chromatic Joy',
  },
  {
    value: '4',
    name: 'Woodland Whisper',
  },
  {
    value: '5',
    name: 'Deep Blush',
  },
];
