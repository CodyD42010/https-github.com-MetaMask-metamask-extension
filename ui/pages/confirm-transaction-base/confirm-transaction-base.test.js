import React from 'react';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { fireEvent } from '@testing-library/react';

import { NetworkType } from '@metamask/controller-utils';
import { NetworkStatus } from '@metamask/network-controller';
import { EthAccountType, EthMethod } from '@metamask/keyring-api';
import {
  TransactionStatus,
  TransactionType,
} from '@metamask/transaction-controller';
import { act } from 'react-dom/test-utils';
import { renderWithProvider } from '../../../test/lib/render-helpers';
import { setBackgroundConnection } from '../../store/background-connection';
import { INITIAL_SEND_STATE_FOR_EXISTING_DRAFT } from '../../../test/jest/mocks';
import { GasEstimateTypes } from '../../../shared/constants/gas';
import { KeyringType } from '../../../shared/constants/keyring';
import {
  CHAIN_IDS,
  GOERLI_DISPLAY_NAME,
  NETWORK_TYPES,
} from '../../../shared/constants/network';
import { domainInitialState } from '../../ducks/domains';

import ConfirmTransactionBase from './confirm-transaction-base.container';

const middleware = [thunk];

setBackgroundConnection({
  getGasFeeTimeEstimate: jest.fn(),
  getGasFeeEstimatesAndStartPolling: jest.fn(),
  promisifiedBackground: jest.fn(),
  tryReverseResolveAddress: jest.fn(),
  getNextNonce: jest.fn(),
});

const mockTxParamsFromAddress = '0x123456789';

const mockTxParamsToAddress = '0x85c1685cfceaa5c0bdb1609fc536e9a8387dd65e';
const mockTxParamsToAddressConcat = '0x85c16...DD65e';

const mockParsedTxDataToAddressWithout0x =
  'e57e7847fd3661a9b7c86aaf1daea08d9da5750a';
const mockParsedTxDataToAddress = '0xe57E7...5750A';

const mockPropsToAddress = '0x33m1685cfceaa5c0bdb1609fc536e9a8387dd567';
const mockPropsToAddressConcat = '0x33m16...dd567';

const mockTxParams = {
  from: mockTxParamsFromAddress,
  to: mockTxParamsToAddress,
  value: '0x5af3107a4000',
  gas: '0x5208',
  maxFeePerGas: '0x59682f16',
  maxPriorityFeePerGas: '0x59682f00',
  type: '0x2',
  data: `0xa22cb465000000000000000000000000${mockParsedTxDataToAddressWithout0x}0000000000000000000000000000000000000000000000000000000000000001`,
};

const baseStore = {
  send: {
    ...INITIAL_SEND_STATE_FOR_EXISTING_DRAFT,
    currentTransactionUUID: null,
    draftTransactions: {},
  },
  DNS: domainInitialState,
  gas: {
    customData: { limit: null, price: null },
  },
  history: { mostRecentOverviewPage: '/' },
  metamask: {
    transactions: [
      {
        id: 1,
        chainId: '0x5',
        txParams: { ...mockTxParams },
        status: 'unapproved',
      },
    ],
    gasEstimateType: GasEstimateTypes.legacy,
    gasFeeEstimates: {
      low: '0',
      medium: '1',
      fast: '2',
    },
    selectedAddress: mockTxParamsFromAddress,
    keyrings: [
      {
        type: KeyringType.hdKeyTree,
        accounts: ['0x0'],
      },
    ],
    selectedNetworkClientId: NetworkType.mainnet,
    networksMetadata: {
      [NetworkType.mainnet]: {
        EIPS: {},
        status: NetworkStatus.Available,
      },
    },
    tokens: [],
    preferences: {
      useNativeCurrencyAsPrimaryCurrency: false,
    },
    currentCurrency: 'USD',
    currencyRates: {},
    providerConfig: {
      chainId: CHAIN_IDS.GOERLI,
      nickname: GOERLI_DISPLAY_NAME,
      type: NETWORK_TYPES.GOERLI,
    },
    featureFlags: {
      sendHexData: false,
    },
    addressBook: {
      [CHAIN_IDS.GOERLI]: [],
    },
    cachedBalances: {
      [CHAIN_IDS.GOERLI]: {},
    },
    accounts: {
      [mockTxParamsFromAddress]: {
        balance: '0x0',
        address: mockTxParamsFromAddress,
      },
    },
    identities: {
      [mockTxParamsFromAddress]: { address: mockTxParamsFromAddress },
      [mockTxParamsToAddress]: {
        name: 'Test Address 1',
      },
    },
    internalAccounts: {
      accounts: {
        'cf8dace4-9439-4bd4-b3a8-88c821c8fcb3': {
          address: mockTxParamsFromAddress,
          id: 'cf8dace4-9439-4bd4-b3a8-88c821c8fcb3',
          metadata: {
            name: 'Account 1',
            keyring: {
              type: 'HD Key Tree',
            },
          },
          options: {},
          methods: [...Object.values(EthMethod)],
          type: EthAccountType.Eoa,
        },
      },
      selectedAccount: 'cf8dace4-9439-4bd4-b3a8-88c821c8fcb3',
    },
    tokenAddress: '0x32e6c34cd57087abbd59b5a4aecc4cb495924356',
    tokenList: {},
    ensResolutionsByAddress: {},
    snaps: {},
  },
  confirmTransaction: {
    txData: {
      id: 1,
      txParams: { ...mockTxParams },
      time: 1675012496170,
      status: TransactionStatus.unapproved,
      originalGasEstimate: '0x5208',
      userEditedGasLimit: false,
      chainId: '0x5',
      loadingDefaults: false,
      dappSuggestedGasFees: null,
      sendFlowHistory: [],
      origin: 'metamask',
      actionId: 1675012496153.2039,
      type: 'simpleSend',
      history: [],
      userFeeLevel: 'medium',
      defaultGasEstimates: {
        estimateType: 'medium',
        gas: '0x5208',
        maxFeePerGas: '0x59682f16',
        maxPriorityFeePerGas: '0x59682f00',
      },
    },
    tokenData: {},
    tokenProps: {},
    fiatTransactionAmount: '0.16',
    fiatTransactionFee: '0',
    fiatTransactionTotal: '0.16',
    ethTransactionAmount: '0.0001',
    ethTransactionFee: '0',
    ethTransactionTotal: '0.0001',
    hexTransactionAmount: '0x5af3107a4000',
    hexTransactionFee: '0x0',
    hexTransactionTotal: '0x5af3107a4000',
    nonce: '',
  },
  appState: {
    sendInputCurrencySwitched: false,
  },
};

const mockedStore = jest.mocked(baseStore);

const mockedStoreWithConfirmTxParams = (_mockTxParams = mockTxParams) => {
  mockedStore.metamask.transactions[0].txParams = { ..._mockTxParams };
  mockedStore.confirmTransaction.txData.txParams = { ..._mockTxParams };
};

const sendToRecipientSelector =
  '.sender-to-recipient__party--recipient .sender-to-recipient__name';

describe('Confirm Transaction Base', () => {
  it('should match snapshot', () => {
    const store = configureMockStore(middleware)(baseStore);
    const { container } = renderWithProvider(
      <ConfirmTransactionBase actionKey="confirm" />,
      store,
    );
    expect(container).toMatchSnapshot();
  });

  it('should not contain L1 L2 fee details for chains that are not optimism', () => {
    const store = configureMockStore(middleware)(baseStore);
    const { queryByText } = renderWithProvider(
      <ConfirmTransactionBase actionKey="confirm" />,
      store,
    );
    expect(queryByText('Layer 1 fees')).not.toBeInTheDocument();
    expect(queryByText('Layer 2 gas fee')).not.toBeInTheDocument();
  });

  it('should contain L1 L2 fee details for optimism', () => {
    mockedStore.metamask.providerConfig.chainId = CHAIN_IDS.OPTIMISM;
    mockedStore.confirmTransaction.txData.chainId = CHAIN_IDS.OPTIMISM;
    const store = configureMockStore(middleware)(mockedStore);
    const { queryByText } = renderWithProvider(
      <ConfirmTransactionBase actionKey="confirm" />,
      store,
    );
    expect(queryByText('Layer 1 fees')).toBeInTheDocument();
    expect(queryByText('Layer 2 gas fee')).toBeInTheDocument();
  });

  it('should render NoteToTrader when isNoteToTraderSupported is true', () => {
    mockedStore.metamask.custodyAccountDetails = {
      [mockTxParamsFromAddress]: {
        address: mockTxParamsFromAddress,
        details: 'details',
        custodyType: 'testCustody - Saturn',
        custodianName: 'saturn-dev',
      },
    };

    mockedStore.metamask.mmiConfiguration = {
      custodians: [
        {
          envName: 'saturn-dev',
          displayName: 'Saturn Custody',
          isNoteToTraderSupported: true,
        },
      ],
    };

    const store = configureMockStore(middleware)(mockedStore);
    const { getByTestId } = renderWithProvider(
      <ConfirmTransactionBase actionKey="confirm" />,
      store,
    );

    expect(getByTestId('note-tab')).toBeInTheDocument();
  });

  it('handleMMISubmit calls sendTransaction correctly when isNoteToTraderSupported is false', async () => {
    const newMockedStore = {
      ...mockedStore,
      appState: {
        ...mockedStore.appState,
        gasLoadingAnimationIsShowing: false,
      },
      confirmTransaction: {
        ...mockedStore.confirmTransaction,
        txData: {
          ...mockedStore.confirmTransaction.txData,
          custodyStatus: true,
        },
      },
      metamask: {
        ...mockedStore.metamask,
        accounts: {
          [mockTxParamsFromAddress]: {
            balance: '0x1000000000000000000',
            address: mockTxParamsFromAddress,
          },
        },
        gasEstimateType: GasEstimateTypes.feeMarket,
        selectedNetworkClientId: NetworkType.mainnet,
        networksMetadata: {
          ...mockedStore.metamask.networksMetadata,
          [NetworkType.mainnet]: {
            EIPS: {
              1559: true,
            },
            status: NetworkStatus.Available,
          },
        },
        customGas: {
          gasLimit: '0x5208',
          gasPrice: '0x59682f00',
        },
        noGasPrice: false,
        keyrings: [
          {
            type: 'Custody',
            accounts: ['0x123456789'],
          },
        ],
        custodyAccountDetails: {
          [mockTxParamsFromAddress]: {
            address: mockTxParamsFromAddress,
            details: 'details',
            custodyType: 'testCustody - Saturn',
            custodianName: 'saturn-dev',
          },
        },
        mmiConfiguration: {
          custodians: [
            {
              envName: 'saturn-dev',
              displayName: 'Saturn Custody',
              isNoteToTraderSupported: false,
            },
          ],
        },
      },
      send: {
        ...mockedStore.send,
        gas: {
          ...mockedStore.send.gas,
          gasEstimateType: GasEstimateTypes.legacy,
          gasFeeEstimates: {
            low: '0',
            medium: '1',
            high: '2',
          },
        },
        hasSimulationError: false,
        userAcknowledgedGasMissing: false,
        submitting: false,
        hardwareWalletRequiresConnection: false,
        gasIsLoading: false,
        gasFeeIsCustom: true,
      },
    };

    const store = configureMockStore(middleware)(newMockedStore);
    const sendTransaction = jest
      .fn()
      .mockResolvedValue(newMockedStore.confirmTransaction.txData);
    const updateTransaction = jest.fn().mockResolvedValue();
    const showCustodianDeepLink = jest.fn();
    const setWaitForConfirmDeepLinkDialog = jest.fn();

    const { getByTestId } = renderWithProvider(
      <ConfirmTransactionBase
        actionKey="confirm"
        sendTransaction={sendTransaction}
        updateTransaction={updateTransaction}
        showCustodianDeepLink={showCustodianDeepLink}
        setWaitForConfirmDeepLinkDialog={setWaitForConfirmDeepLinkDialog}
        toAddress={mockPropsToAddress}
        toAccounts={[{ address: mockPropsToAddress }]}
        isMainBetaFlask={false}
      />,
      store,
    );

    const confirmButton = getByTestId('page-container-footer-next');

    await act(async () => {
      fireEvent.click(confirmButton);
    });

    expect(sendTransaction).toHaveBeenCalled();
  });

  it('handleMainSubmit calls sendTransaction correctly', async () => {
    const newMockedStore = {
      ...mockedStore,
      appState: {
        ...mockedStore.appState,
        gasLoadingAnimationIsShowing: false,
      },
      metamask: {
        ...mockedStore.metamask,
        accounts: {
          [mockTxParamsFromAddress]: {
            balance: '0x1000000000000000000',
            address: mockTxParamsFromAddress,
          },
        },
        gasEstimateType: GasEstimateTypes.feeMarket,
        selectedNetworkClientId: NetworkType.mainnet,
        networksMetadata: {
          ...mockedStore.metamask.networksMetadata,
          [NetworkType.mainnet]: {
            EIPS: { 1559: true },
            status: NetworkStatus.Available,
          },
        },
        customGas: {
          gasLimit: '0x5208',
          gasPrice: '0x59682f00',
        },
        noGasPrice: false,
      },
      send: {
        ...mockedStore.send,
        gas: {
          ...mockedStore.send.gas,
          gasEstimateType: GasEstimateTypes.legacy,
          gasFeeEstimates: {
            low: '0',
            medium: '1',
            high: '2',
          },
        },
        hasSimulationError: false,
        userAcknowledgedGasMissing: false,
        submitting: false,
        hardwareWalletRequiresConnection: false,
        gasIsLoading: false,
        gasFeeIsCustom: true,
      },
    };
    const store = configureMockStore(middleware)(newMockedStore);
    const sendTransaction = jest.fn().mockResolvedValue();

    const { getByTestId } = renderWithProvider(
      <ConfirmTransactionBase
        actionKey="confirm"
        sendTransaction={sendTransaction}
        toAddress={mockPropsToAddress}
        toAccounts={[{ address: mockPropsToAddress }]}
      />,
      store,
    );
    const confirmButton = getByTestId('page-container-footer-next');
    fireEvent.click(confirmButton);
    expect(sendTransaction).toHaveBeenCalled();
  });

  it('handleMMISubmit calls sendTransaction correctly and then showCustodianDeepLink', async () => {
    const newMockedStore = {
      ...mockedStore,
      appState: {
        ...mockedStore.appState,
        gasLoadingAnimationIsShowing: false,
      },
      confirmTransaction: {
        ...mockedStore.confirmTransaction,
        txData: {
          ...mockedStore.confirmTransaction.txData,
          custodyStatus: true,
        },
      },
      metamask: {
        ...mockedStore.metamask,
        accounts: {
          [mockTxParamsFromAddress]: {
            balance: '0x1000000000000000000',
            address: mockTxParamsFromAddress,
          },
        },
        gasEstimateType: GasEstimateTypes.feeMarket,
        selectedNetworkClientId: NetworkType.mainnet,
        networksMetadata: {
          ...mockedStore.metamask.networksMetadata,
          [NetworkType.mainnet]: {
            EIPS: {
              1559: true,
            },
            status: NetworkStatus.Available,
          },
        },
        customGas: {
          gasLimit: '0x5208',
          gasPrice: '0x59682f00',
        },
        noGasPrice: false,
      },
      send: {
        ...mockedStore.send,
        gas: {
          ...mockedStore.send.gas,
          gasEstimateType: GasEstimateTypes.legacy,
          gasFeeEstimates: {
            low: '0',
            medium: '1',
            high: '2',
          },
        },
        hasSimulationError: false,
        userAcknowledgedGasMissing: false,
        submitting: false,
        hardwareWalletRequiresConnection: false,
        gasIsLoading: false,
        gasFeeIsCustom: true,
      },
    };
    const store = configureMockStore(middleware)(newMockedStore);
    const sendTransaction = jest
      .fn()
      .mockResolvedValue(newMockedStore.confirmTransaction.txData);
    const showCustodianDeepLink = jest.fn();
    const setWaitForConfirmDeepLinkDialog = jest.fn();

    const { getByTestId } = renderWithProvider(
      <ConfirmTransactionBase
        actionKey="confirm"
        sendTransaction={sendTransaction}
        showCustodianDeepLink={showCustodianDeepLink}
        setWaitForConfirmDeepLinkDialog={setWaitForConfirmDeepLinkDialog}
        toAddress={mockPropsToAddress}
        toAccounts={[{ address: mockPropsToAddress }]}
        isMainBetaFlask={false}
      />,
      store,
    );
    const confirmButton = getByTestId('page-container-footer-next');
    fireEvent.click(confirmButton);
    expect(setWaitForConfirmDeepLinkDialog).toHaveBeenCalled();
    await expect(sendTransaction).toHaveBeenCalled();
    expect(showCustodianDeepLink).toHaveBeenCalled();
  });

  describe('when rendering the recipient value', () => {
    describe(`when the transaction is a ${TransactionType.simpleSend} type`, () => {
      it(`should use txParams.to address`, () => {
        const store = configureMockStore(middleware)(mockedStore);
        const { container } = renderWithProvider(
          <ConfirmTransactionBase actionKey="confirm" />,
          store,
        );

        const recipientElem = container.querySelector(sendToRecipientSelector);
        expect(recipientElem).toHaveTextContent(mockTxParamsToAddressConcat);
      });

      it(`should use txParams.to address even if there is no amount sent`, () => {
        mockedStoreWithConfirmTxParams({
          ...mockTxParams,
          value: '0x0',
        });
        const store = configureMockStore(middleware)(mockedStore);
        const { container } = renderWithProvider(
          <ConfirmTransactionBase actionKey="confirm" />,
          store,
        );

        const recipientElem = container.querySelector(sendToRecipientSelector);
        expect(recipientElem).toHaveTextContent(mockTxParamsToAddressConcat);
      });
    });
    describe(`when the transaction is NOT a ${TransactionType.simpleSend} type`, () => {
      beforeEach(() => {
        mockedStore.confirmTransaction.txData.type =
          TransactionType.contractInteraction;
      });

      describe('when there is an amount being sent (it should be treated as a general contract intereaction rather than custom one)', () => {
        it('should use txParams.to address (contract address)', () => {
          mockedStoreWithConfirmTxParams({
            ...mockTxParams,
            value: '0x45666',
          });
          const store = configureMockStore(middleware)(mockedStore);
          const { container } = renderWithProvider(
            <ConfirmTransactionBase actionKey="confirm" />,
            store,
          );

          const recipientElem = container.querySelector(
            sendToRecipientSelector,
          );
          expect(recipientElem).toHaveTextContent(mockTxParamsToAddressConcat);
        });
      });

      describe(`when there is no amount being sent`, () => {
        it('should use propToAddress (toAddress passed as prop)', () => {
          mockedStoreWithConfirmTxParams({
            ...mockTxParams,
            value: '0x0',
          });
          const store = configureMockStore(middleware)(mockedStore);

          const { container } = renderWithProvider(
            <ConfirmTransactionBase
              // we want to test toAddress provided by ownProps in mapStateToProps, but this
              // currently overrides toAddress this should pan out fine when we refactor the
              // component into a functional component and remove the container.js file
              toAddress={mockPropsToAddress}
              actionKey="confirm"
            />,
            store,
          );

          const recipientElem = container.querySelector(
            sendToRecipientSelector,
          );
          expect(recipientElem).toHaveTextContent(mockPropsToAddressConcat);
        });

        it('should use address parsed from transaction data if propToAddress is not provided', () => {
          mockedStoreWithConfirmTxParams({
            ...mockTxParams,
            value: '0x0',
          });
          const store = configureMockStore(middleware)(mockedStore);
          const { container } = renderWithProvider(
            <ConfirmTransactionBase actionKey="confirm" />,
            store,
          );

          const recipientElem = container.querySelector(
            sendToRecipientSelector,
          );
          expect(recipientElem).toHaveTextContent(mockParsedTxDataToAddress);
        });

        it('should use txParams.to if neither propToAddress is not provided nor the transaction data to address were provided', () => {
          mockedStoreWithConfirmTxParams({
            ...mockTxParams,
            data: '0x',
            value: '0x0',
          });
          const store = configureMockStore(middleware)(mockedStore);
          const { container } = renderWithProvider(
            <ConfirmTransactionBase actionKey="confirm" />,
            store,
          );

          const recipientElem = container.querySelector(
            sendToRecipientSelector,
          );
          expect(recipientElem).toHaveTextContent(mockTxParamsToAddressConcat);
        });
      });
    });
  });
});
