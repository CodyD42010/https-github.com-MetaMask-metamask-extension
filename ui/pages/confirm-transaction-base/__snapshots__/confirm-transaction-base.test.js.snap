// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Transaction Base should match snapshot 1`] = `
<div>
  <div
    class="page-container"
    data-testid="page-container"
  >
    <div
      class="confirm-page-container-navigation"
      style="display: none;"
    >
      <div
        class="confirm-page-container-navigation__container"
        data-testid="navigation-container"
        style="visibility: hidden;"
      >
        <button
          class="confirm-page-container-navigation__arrow"
          data-testid="first-page"
        >
          <i
            class="fa fa-angle-double-left fa-2x"
          />
        </button>
        <button
          class="confirm-page-container-navigation__arrow"
          data-testid="previous-page"
        >
          <i
            class="fa fa-angle-left fa-2x"
          />
        </button>
      </div>
      <div
        class="confirm-page-container-navigation__textcontainer"
      >
        <div
          class="confirm-page-container-navigation__navtext"
        >
          0
           
          of
           
          1
        </div>
        <div
          class="confirm-page-container-navigation__longtext"
        >
          requests waiting to be acknowledged
        </div>
      </div>
      <div
        class="confirm-page-container-navigation__container"
        style="visibility: initial;"
      >
        <button
          class="confirm-page-container-navigation__arrow"
          data-testid="next-page"
        >
          <i
            class="fa fa-angle-right fa-2x"
          />
        </button>
        <button
          class="confirm-page-container-navigation__arrow"
          data-testid="last-page"
        >
          <i
            class="fa fa-angle-double-right fa-2x"
          />
        </button>
      </div>
    </div>
    <div
      class="sender-to-recipient sender-to-recipient--default"
      data-testid="sender-to-recipient"
    >
      <div
        class="sender-to-recipient__party sender-to-recipient__party--sender"
      >
        <div
          class="sender-to-recipient__sender-icon"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 24px; width: 24px; border-radius: 12px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 24px; height: 24px; display: inline-block; background: rgb(245, 228, 0);"
              >
                <svg
                  height="24"
                  width="24"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#F2BE02"
                    height="24"
                    transform="translate(4.11204385070503 -3.89652025148055) rotate(387.1 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#01778E"
                    height="24"
                    transform="translate(-1.9540919563657109 12.528454691642345) rotate(227.4 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#C81435"
                    height="24"
                    transform="translate(13.477268130678231 -9.444486788922836) rotate(422.3 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sender-to-recipient__tooltip-wrapper"
        >
          <div
            aria-describedby="tippy-tooltip-1"
            class="sender-to-recipient__tooltip-container"
            data-original-title="null"
            data-tooltipped=""
            style="display: inline;"
            tabindex="0"
          >
            <div
              class="sender-to-recipient__name"
            />
          </div>
        </div>
      </div>
      <div
        class="sender-to-recipient__arrow-container"
      >
        <div
          class="sender-to-recipient__arrow-circle"
        >
          <i
            class="fa fa-arrow-right sender-to-recipient__arrow-circle__icon"
          />
        </div>
      </div>
      <div
        class="sender-to-recipient__party sender-to-recipient__party--recipient sender-to-recipient__party--recipient-with-address"
      >
        <div
          class="sender-to-recipient__sender-icon"
        >
          <div
            class=""
          >
            <div
              class="identicon"
              style="height: 24px; width: 24px; border-radius: 12px;"
            >
              <div
                style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 24px; height: 24px; display: inline-block; background: rgb(249, 108, 1);"
              >
                <svg
                  height="24"
                  width="24"
                  x="0"
                  y="0"
                >
                  <rect
                    fill="#2343E1"
                    height="24"
                    transform="translate(1.995636908144359 -2.9078959833795546) rotate(391.8 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#F2CA02"
                    height="24"
                    transform="translate(-10.28190700465709 0.16715759851871784) rotate(278.6 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                  <rect
                    fill="#C8142C"
                    height="24"
                    transform="translate(16.427173764273576 -13.297284285395893) rotate(345.3 12 12)"
                    width="24"
                    x="0"
                    y="0"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sender-to-recipient__tooltip-wrapper"
        >
          <div
            class="sender-to-recipient__tooltip-container"
            style="display: inline;"
            tabindex="0"
            title=""
          >
            <div
              class="sender-to-recipient__name"
              data-testid="sender-to-recipient__name"
            >
              0x85c...D65e
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="confirm-page-container-content"
    >
      
      <div
        class="confirm-page-container-summary"
      >
        <div
          class="confirm-page-container-summary__action-row"
        >
          <div
            class="confirm-page-container-summary__action"
          >
            <span
              class="confirm-page-container-summary__action__name"
            >
              Sending ETH
            </span>
          </div>
          <div
            class="confirm-page-container-summary__nonce"
          >
            #undefined
          </div>
        </div>
        <div
          class="confirm-page-container-summary__title"
        >
          <h3
            class="box mm-text mm-text--heading-md mm-text--font-weight-normal mm-text--ellipsis box--flex-direction-row box--color-text-default"
          >
            <div
              class="currency-display-component"
              title="0.0001"
            >
              <span
                class="currency-display-component__prefix"
              >
                <i
                  class="fab fa-ethereum"
                  style="font-size: 24px;"
                />
              </span>
              <span
                class="currency-display-component__text"
              >
                0.0001
              </span>
            </div>
          </h3>
        </div>
      </div>
      <div
        class="box tabs box--flex-direction-row"
      >
        <ul
          class="box tabs__list box--display-flex box--gap-1 box--flex-direction-row box--justify-content-flex-start box--background-color-background-default"
        >
          <li
            class="box tab confirm-page-container-content__tab tab--active box--flex-direction-row"
          >
            <button
              class="box box--padding-2 box--display-block box--flex-direction-row box--text-align-center box--width-full"
            >
              Details
            </button>
          </li>
          <li
            class="box tab confirm-page-container-content__tab box--flex-direction-row"
          >
            <button
              class="box box--padding-2 box--display-block box--flex-direction-row box--text-align-center box--width-full"
            >
              Data
            </button>
          </li>
          <li
            class="box tab confirm-page-container-content__tab box--flex-direction-row"
          >
            <button
              class="box box--padding-2 box--display-block box--flex-direction-row box--text-align-center box--width-full"
            >
              Hex
            </button>
          </li>
        </ul>
        <div
          class="box tabs__content box--flex-direction-row"
        >
          <div
            class="confirm-page-container-content__details"
          >
            <div
              class="transaction-alerts"
            />
            <div
              class="transaction-detail"
            >
              <div
                class="transaction-detail-edit"
              >
                <button>
                  Edit
                </button>
              </div>
              <div
                class="transaction-detail-rows"
              >
                <div
                  class="transaction-detail-item"
                >
                  <div
                    class="transaction-detail-item__row"
                  >
                    <h6
                      class="box box--margin-top-1 box--margin-bottom-1 box--display-flex box--flex-direction-row box--flex-wrap-nowrap box--align-items-center typography typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
                    >
                      Estimated gas fee
                      <div
                        class="info-tooltip"
                        data-testid="info-tooltip"
                      >
                        <div>
                          <div
                            aria-describedby="tippy-tooltip-2"
                            class="info-tooltip__tooltip-container"
                            data-original-title="null"
                            data-tooltipped=""
                            style="display: inline;"
                            tabindex="0"
                          >
                            <svg
                              viewBox="0 0 10 10"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <path
                                d="M5 0C2.2 0 0 2.2 0 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 2c.4 0 .7.3.7.7s-.3.7-.7.7-.7-.2-.7-.6.3-.8.7-.8zm.7 6H4.3V4.3h1.5V8z"
                                fill="var(--color-icon-alternative)"
                              />
                            </svg>
                          </div>
                        </div>
                      </div>
                    </h6>
                    <div
                      class="transaction-detail-item__detail-values"
                    >
                      <h6
                        class="box box--margin-top-1 box--margin-bottom-1 box--margin-left-1 box--flex-direction-row box--text-align-right typography typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
                      >
                        <div>
                          <div
                            class="currency-display-component"
                            title="0.000021"
                          >
                            <span
                              class="currency-display-component__prefix"
                            />
                            <span
                              class="currency-display-component__text"
                            >
                              0.000021
                            </span>
                          </div>
                        </div>
                      </h6>
                    </div>
                  </div>
                  <div
                    class="transaction-detail-item__row"
                  >
                    <div />
                    <h6
                      class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography transaction-detail-item__row-subText typography--h7 typography--weight-normal typography--style-normal typography--align-end typography--color-text-alternative"
                    >
                      <strong>
                        Max fee:
                      </strong>
                      <div>
                        <div
                          class="currency-display-component"
                          title="0.000021"
                        >
                          <span
                            class="currency-display-component__prefix"
                          />
                          <span
                            class="currency-display-component__text"
                          >
                            0.000021
                          </span>
                        </div>
                      </div>
                    </h6>
                  </div>
                </div>
                <div
                  class="transaction-detail-item"
                >
                  <div
                    class="transaction-detail-item__row"
                  >
                    <h6
                      class="box box--margin-top-1 box--margin-bottom-1 box--display-flex box--flex-direction-row box--flex-wrap-nowrap box--align-items-center typography typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
                    >
                      Total
                    </h6>
                    <div
                      class="transaction-detail-item__detail-values"
                    >
                      <h6
                        class="box box--margin-top-1 box--margin-bottom-1 box--margin-left-1 box--flex-direction-row box--text-align-right typography typography--h6 typography--weight-bold typography--style-normal typography--color-text-default"
                      >
                        <div
                          class="confirm-page-container-content__total-value"
                        >
                          <div
                            class="currency-display-component"
                            title="0.000121"
                          >
                            <span
                              class="currency-display-component__prefix"
                            />
                            <span
                              class="currency-display-component__text"
                            >
                              0.000121
                            </span>
                          </div>
                        </div>
                      </h6>
                    </div>
                  </div>
                  <div
                    class="transaction-detail-item__row"
                  >
                    <h6
                      class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography typography--h7 typography--weight-normal typography--style-normal typography--color-text-alternative"
                    >
                      Amount + gas fee
                    </h6>
                    <h6
                      class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography transaction-detail-item__row-subText typography--h7 typography--weight-normal typography--style-normal typography--align-end typography--color-text-alternative"
                    >
                      <div
                        class="confirm-page-container-content__total-amount"
                      >
                        <strong>
                          Max amount:
                        </strong>
                         
                        <div
                          class="currency-display-component"
                          title="0.000121"
                        >
                          <span
                            class="currency-display-component__prefix"
                          />
                          <span
                            class="currency-display-component__text"
                          >
                            0.000121
                          </span>
                        </div>
                      </div>
                    </h6>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="confirm-page-container-content__error-container"
      >
        <div
          class="actionable-message actionable-message--danger actionable-message--warning actionable-message--with-icon"
        >
          <svg
            viewBox="0 0 10 10"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 0C2.2 0 0 2.2 0 5s2.2 5 5 5 5-2.2 5-5-2.2-5-5-5zm0 2c.4 0 .7.3.7.7s-.3.7-.7.7-.7-.2-.7-.6.3-.8.7-.8zm.7 6H4.3V4.3h1.5V8z"
              fill="var(--color-error-default)"
            />
          </svg>
          
          <div
            class="actionable-message__message"
          >
            <h6
              class="box box--margin-top-1 box--margin-bottom-1 box--flex-direction-row typography typography--h7 typography--weight-normal typography--style-normal typography--align-left typography--color-text-default"
            >
              You do not have enough  in your account to pay for transaction fees on Goerli network. Deposit  from another account.
            </h6>
          </div>
        </div>
      </div>
      <div
        class="page-container__footer"
      >
        <footer>
          <button
            class="button btn--rounded btn-secondary page-container__footer-button page-container__footer-button__cancel"
            data-testid="page-container-footer-cancel"
            role="button"
            tabindex="0"
          >
            Reject
          </button>
          <button
            class="button btn--rounded btn-primary page-container__footer-button"
            data-testid="page-container-footer-next"
            disabled=""
            role="button"
            tabindex="0"
          >
            Confirm
          </button>
        </footer>
      </div>
    </div>
  </div>
</div>
`;
